merge/schema_change_test.go

```diff
@@ -252,3 +252,459 @@ func TestAtomicToGranularSchemaChanges(t *testing.T) {
 		})
 	}
 }
+
+var associativeListParserOld = func() *typed.Parser {
+	oldParser, err := typed.NewParser(`types:
+- name: v1
+  map:
+    fields:
+      - name: list
+        type:
+          namedType: associativeList
+- name: associativeList
+  list:
+    elementType:
+      namedType: myElement
+    elementRelationship: associative
+    keys:
+    - name
+- name: myElement
+  map:
```

Comment:
do we have to include id here for old data to persist, even if we didn't make it a list key?

Reply:
I split the tests into two groups. The ones here in `TestAssociativeListSchemaChanges` simulate adding a new key field.  The ones in below in `TestPromoteFieldToAssociativeListKey` simulate promoting an existing field to a key.

Reply:
ah, ok, I thought this schema was used as the starting point for both tests